docs/painless/painless-guide/painless-datetime.asciidoc

```diff
@@ -0,0 +1,295 @@
+long milliSinceEpoch = zdt.toInstant().toEpochMilli();
+----
++
+* format to a String (ISO 1806)
++
+[source,Painless]
+----
+ZonedDateTime zdt =
+        ZonedDateTime.of(1983, 10, 13, 22, 15, 30, 0, ZoneId.of('Z'));
+String datetime = zdt.format(DateTimeFormatter.ISO_INSTANT);
+----
+Note the specification of a built-in DateTimeFormatter.
++
+* format to a String (custom)
++
+[source,Painless]
+----
+ZonedDateTime zdt =
+        ZonedDateTime.of(1983, 10, 13, 22, 15, 30, 0, ZoneId.of('Z'));
+DateTimeFormatter dtf = DateTimeFormatter.ofPattern(
+        "'date:' yyyy/MM/dd 'time:' HH:mm:ss");
+String datetime = zdt.format(dtf);
+----
+Note the specification of a custom DateTimeFormatter.
++
+* format to individual pieces
++
+[source,Painless]
+----
+ZonedDateTime zdt =
+        ZonedDateTime.of(1983, 10, 13, 22, 15, 30, 100, ZoneId.of(tz));
+int year = zdt.year;
+int month = zdt.monthValue;
+int day = zdt.dayOfMonth;
+int hour = zdt.hour;
+int minutes = zdt.minute;
+int seconds = zdt.second;
+int nanos = zdt.nano;
+----
+
+==== Datetime Modification
+
+Modifications to datetimes such as adding/subtracting a few seconds or setting
+a specific day are often required to meet a script's objectives. Use either a
+long or a ZonedDateTime to do datetime modification. Use standard
+<<painless-operators-numeric, numeric operators>> to modify a long
+representation of a datetime. Use <<method-call-operator, method calls>> to
+modify a ZonedDateTime representation of a datetime. Note most modification
+methods for a ZonedDateTime return a new instance that needs assignment or
+immediate use.
+
+===== Datetime Modification Examples
+
+* Subtract three seconds from a long (milliseconds)
++
+[source,Painless]
+----
+long milliSinceEpoch = 434931330000L;
+milliSinceEpoch = milliSinceEpoch - 1000L*3L;
+----
++
+* Add three days to a datetime
++
+[source,Painless]
+----
+ZonedDateTime zdt =
+        ZonedDateTime.of(1983, 10, 13, 22, 15, 30, 0, ZoneId.of('Z'));
+ZonedDateTime updatedZdt = zdt.plusDays(3);
+----
++
+* Subtract 125 minutes from a datetime
++
+[source,Painless]
+----
+ZonedDateTime zdt =
+        ZonedDateTime.of(1983, 10, 13, 22, 15, 30, 0, ZoneId.of('Z'));
+ZonedDateTime updatedZdt = zdt.minusMinutes(125);
+----
++
+* Set the year on a datetime
++
+[source,Painless]
+----
+ZonedDateTime zdt =
+        ZonedDateTime.of(1983, 10, 13, 22, 15, 30, 0, ZoneId.of('Z'));
+ZonedDateTime updatedZdt = zdt.withYear(1976);
+----
+
+==== Elapsed Time
+
+A script's objectives may require the elapsed time (difference) between two
+datetimes. Use either two longs or two ZonedDateTimes to calculate an elapsed
+time. Use <<subtraction-operator, subtraction>> to calculate an elapsed time
+between two longs. Use <<painless-api-reference-shared-ChronoUnit, ChronoUnit>>
+to calculate the difference between two ZonedDateTimes.
+
+===== Elapsed Time Examples
+
+* Elapsed time for two longs (milliseconds)
```

Comment:
This seems unnecessary, as it is just teaching to do subtraction of epoch milli values?

Reply:
So I would actually like to leave this example for the following reasons:
1. There are very few really simple examples in the documentation and that's been one of the primary complaints I've seen.
2. Many of our users may not have a programming background and just want to get something to work without having to develop a huge knowledge base.  This kind of example is something that helps a lot here.
3. Advanced users that don't need the example can quickly skip this section.